File: flutter_app/lib/core/models/action_log.dart
class ActionLog {
  final String id;
  final String userId;
  final String action;
  final String description;
  final DateTime createdAt;

  ActionLog({
    required this.id,
    required this.userId,
    required this.action,
    required this.description,
    required this.createdAt,
  });

  factory ActionLog.fromJson(Map<String, dynamic> json) {
    return ActionLog(
      id: json['id'].toString(),
      userId: json['user_action_logs'] ?? '', 
      action: json['action'] ?? '',
      description: json['description'] ?? '',
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toString()),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_action_logs': userId, 
      'action': action,
      'description': description,
      'created_at': createdAt.toIso8601String(),
    };
  }
}

File: golang_app/entity/actionlog.go
package entity

import (
	"time"

	"github.com/asma12a/challenge-s6/ent"
	"github.com/asma12a/challenge-s6/ent/schema/ulid"
)

type ActionLog struct {
	ent.ActionLog
	UserID    *ulid.ID  `json:"user_id,omitempty"`
	CreatedAt time.Time `json:"created_at"`
}

func NewActionLog(
	userID *ulid.ID, action string, description string,
) *ActionLog {
	return &ActionLog{
		ActionLog: ent.ActionLog{
			Action:      action,
			Description: description,
			CreatedAt:   time.Now(),
		},
		UserID: userID,
	}
}
